In [1]:
import pandas as pd
import numpy as np
import math
from scipy import stats

sigma1 and sigma2 are known

In [2]:
def zp(x1,x2,sigma1,sigma2,n1,n2):
    z = (x1-x2)/(math.sqrt(((sigma1**2)/n1)+((sigma2**2)/n2)))
    if(z<0):
        p = stats.norm.cdf(z) # lhs area
    else:
        p = 1-stats.norm.cdf(z) # rhs area
    
    print(z,p)

In [3]:
zp(121,112,8,8,10,10)

2.5155764746872635 0.00594189462107364


sigma1^2 and sigma2^2 unknown, assumed equal

Lec-20 25:52

In [4]:
b = [89.19,90.05,90.46,93.21,97.19,97.04,91.07,92.75]
a = [91.5,94.18,92.18,95.39,91.79,89.07,94.72,89.21]

In [5]:
stats.ttest_ind(a,b,equal_var = True)

Ttest_indResult(statistic=-0.26524493759007167, pvalue=0.7946864078466069)

In [6]:
stats.t.ppf(0.025,14) # critical t-value 
# alpha/2, n1+n2-2

-2.1447866879169277

t-value is -2.14 and our test statsistics is -0.35 lying on acceptance region

Summary
3 Types of problems:
1) sigma1^2 and sigma2^2 known: compared to mean           
2) sigma1^2 and sigma2^2 unknown and equal variance: Z&T-Test           
3) sigma1^2 and sigma2^2 unknown and not equal variance: Lec-21            

If you want to know diff bw 2 population variance: add 2 variance


# Lec-21

In [8]:
stats.t.ppf(0.025,13) # critical t-value (area on lhs, degrees of freedom)

-2.160368656461013

In [9]:
metro = [3,7,25,10,15,6,12,25,15,7]
rural = [48,44,40,38,33,21,20,12,1,18]

In [10]:
stats.ttest_ind(metro,rural,equal_var=False) # 2-tailed test 
# reject null hypothesis, p-value compared to alpha is very small

Ttest_indResult(statistic=-2.7669395785560553, pvalue=0.0158272848161009)

In [11]:
# Inferences about the diff bw 2 population means: matched samples
KARL = [1.186,1.151,1.322,1.339,1.200,1.402,1.365,1.537,1.559]
LEH = [1.061,0.992,1.063,1.062,1.065,1.178,1.037,1.086,1.052]

In [12]:
stats.ttest_rel(KARL,LEH)
# _rel means relative so that dependent sample so called the variable 
# you will get t-value and p-value<alpha So reject null hypothesis

Ttest_relResult(statistic=6.0819394375848255, pvalue=0.00029529546278604066)

In [13]:
# Lec-21 21:30
import math
def two_samp_prop(p1,p2,n1,n2):
    p_pool = ((p1*n1)+(p2*n2))/(n1+n2) # pooled proportion
    x = (p_pool*(1-p_pool)*((1/n1)+(1/n2)))
    s = math.sqrt(x)
    z = (p1-p2)/s
    if(z<0):
        p_val = stats.norm.cdf(z)
    else:
        p_val = 1 - stats.norm.cdf(z)
    return z, p_val*2

In [14]:
two_samp_prop(0.27,0.19,100,100)
# Output: t-value,p-value 
# Accept null hypothesis

(1.3442056254198995, 0.17888190308175567)

In [15]:
# z-critical value
stats.norm.cdf(1.3442056254198995)

0.9105590484591222

Next lecture: 2 population variance

# Lec-22

In [16]:
import scipy

In [17]:
# Lec-22 11:12 or 
# upper limit
scipy.stats.f.ppf(q=1-0.05, dfn=15, dfd=10)
# dfd: denominator degrees of freedom
# q: probability
# dfn: numerated degrees of freedom

2.8450165269958436

In [18]:
# lower limit
scipy.stats.f.ppf(q=0.05, dfn=15, dfd=10)

0.3931252536255495

F-Test Example

In [19]:
# X & Y 2 populations are given
# Null Hypothesis: H0 -> (sigma(x))^2=(sigma(y))^2
# Alternative Hypothesis: H1 -> (sigma(x))^2 != (sigma(y))^2
# alpha = 5% = 0.05
X = [3,7,25,10,15,6,12,25,15,7]
Y = [48,44,40,48,33,21,20,12,1,18]
# import numpy if not imported

In [20]:
# Ratio
F = np.var(X)/np.var(Y)
# numerator degrees of freedom
dfn = len(X) - 1 #len(X) tell you no. of elements in array and -1 to get degrees of freedom
# denominator degrees of freedom
dfd = len(Y) - 1

In [21]:
p_value = scipy.stats.f.cdf(F,dfn,dfd)
p_value

0.016274823086223054

Whenever degrees of freedom is increasing it behave like z-distribution ie, when mountainuous graph is 'narrow and high (z-test behavior)' not 'broad and short (t-test behavior)' 

In [22]:
# Lec-22 22:19 or ScreenShot(877)
# pandas, numpy, stats from scipy and math
def samplesize(alpha,beta,mu1,mu2,sigma):
    z1 = -1*stats.norm.ppf(alpha)
    z2 = -1*stats.norm.ppf(beta)
    n = (((z1+z2)**2)*(sigma**2))/((mu1-mu2)**2)
    # z1 is sigma1 and z2 is sigma2
    print(n)

In [23]:
samplesize(0.05,0.1,12,12.75,3.2)

155.900083325938


Compared 2 populations
Compared Variance of 2 population
We have tested whether populations are variants or equal or not
When to go for z and when for t test
Then by considering alpha and beta value we found formula how to derve the value of sample size it would take an ld sample problem then we found what was the value of sample size